### ISSUE TITLE
roslaunch_add_file_check does not consider test_depend, only on buildfirm

### ISSUE THREAD

**Issue by k-okada** [ros-infrastructure/ros_buildfarm#341]
Hi

I have received email from build from about test failure
http://build.ros.org/job/Idev__opencv_apps__ubuntu_trusty_amd64/7/consoleFull#console-section-17
```
19:51:33 Scanning dependencies of target _run_tests_opencv_apps_roslaunch-check__tmp_catkin_workspace_src_opencv_apps_test_test-face_detection.test
19:51:34 -- run_tests.py: execute commands
19:51:34   /usr/bin/cmake -E make_directory /tmp/catkin_workspace/test_results/opencv_apps
19:51:34   /opt/ros/indigo/share/roslaunch/cmake/../scripts/roslaunch-check -o /tmp/catkin_workspace/test_results/opencv_apps/roslaunch-check__tmp_catkin_workspace_src_opencv_apps_test_test-face_detection.test.xml /tmp/catkin_workspace/src/opencv_apps/test/test-face_detection.test 
19:51:34 checking /tmp/catkin_workspace/src/opencv_apps/test/test-face_detection.test
19:51:34 the rosdep view is empty: call 'sudo rosdep init' and 'rosdep update'
19:51:34 Missing package dependencies: opencv_apps/package.xml: image_proc, rostest, rosbag, image_view
19:51:34 ...writing test results to /tmp/catkin_workspace/test_results/opencv_apps/roslaunch-check__tmp_catkin_workspace_src_opencv_apps_test_test-face_detection.test.xml
19:51:34 FAILURE:
19:51:34 [/tmp/catkin_workspace/src/opencv_apps/test/test-face_detection.test]:
19:51:34 	Missing package dependencies: opencv_apps/package.xml: image_proc, rostest, rosbag, image_view
19:51:34 wrote test file to [/tmp/catkin_workspace/test_results/opencv_apps/roslaunch-check__tmp_catkin_workspace_src_opencv_apps_test_test-face_detection.test.xml]
19:51:34 -- run_tests.py: verify result "/tmp/catkin_workspace/test_results/opencv_apps/roslaunch-check__tmp_catkin_workspace_src_opencv_apps_test_test-face_detection.test.xml"
19:51:34 [100%] Built target _run_tests_opencv_apps_roslaunch-check__tmp_catkin_workspace_src_opencv_apps_test_test-face_detection.test
```
it basically notify us that the package.xml does not contain rostest depends.
But actually, I have added them as `test_depend` https://github.com/ros-perception/opencv_apps/blob/indigo/package.xml#L32-L39

It does not happens on our travis which uses `catkin` command https://travis-ci.org/ros-perception/opencv_apps/jobs/156398335 and I have also tested with `catkin_make_isolated` command on my local machine, but could not reproduce this. 
So does ros_buildfarm has different setup as we use catkin_make_isolated in local environment?

**Comment by dirk-thomas** [comment]
The Travis jobs has the same messages ("Missing package dependencies")  as the Jenkins build but with the suffix "notify upstream maintainer". It seems the build is different which affect the classification of the warnings: https://github.com/ros/ros_comm/blob/c37ce07f12bb3361653aea298a2f3898aacc818a/tools/roslaunch/src/roslaunch/rlutil.py#L220-L224 Therefore one fails and the other doesn't. I suspect it is due to the fact that the Jenkins build doesn't provide a rosdep database while your Travis job does. See https://github.com/ros/ros_comm/blob/c37ce07f12bb3361653aea298a2f3898aacc818a/tools/roslaunch/src/roslaunch/rlutil.py#L212-L219

Btw. that logic was introduced by you some time ago :wink: https://github.com/ros/ros_comm/commit/b76a7f3e498b24bd67482451c846df5bb7e6ec84

* Should the test gracefully fall back when it can't use rosdep?
* Should the buildfarm provide a rosdep database for each devel / pr job?

**Comment by k-okada** [comment]
> Btw. that logic was introduced by you some time ago :wink: ros/ros_comm@b76a7f3

oh... I didn't know that...
Looking at the source code, I think the problems is that the `roslaunch_deps` (https://github.com/ros/ros_comm/blob/c37ce07f12bb3361653aea298a2f3898aacc818a/tools/roslaunch/src/roslaunch/depends.py#L319) returns packages within `test_depends` as missing, because of https://github.com/ros/ros_comm/pull/885

As for
> Should the test gracefully fall back when it can't use rosdep?
> Should the buildfarm provide a rosdep database for each devel / pr job?

This a difficult question, I do no t  know how the lack of rosdep database affects `roslaunch-check` but it should run without the database, because roslauch-check only existence of depends, not try to find the actual packages.

**Comment by dirk-thomas** [comment]
This should be fixed with ros/ros_comm#910.
